agent/consul/auto_config_endpoint.go

```diff
@@ -283,6 +283,9 @@ func (ac *AutoConfig) updateTLSSettingsInConfig(_ AutoConfigOptions, resp *pbaut
 	}
 
 	var err error
+
+	// FIXME: is the response from this not actually validated in
+	// auto_config_endpoint_test.go?
```

Comment:
I think it is and I forgot to move the tests during refactoring #12504 🙈

Reply:
Looks like it's tested now after most recent rebase!